### PR TITLE
Granicus site

### PR DESCRIPTION
**Pull request by fatima3558**
This PR adds a `GranicusSite` class

Handles #13 

## Testing
* run `python -m tests.granicus_site` and make sure there are no assertion errors

### CODE REVIEW

**Comment by fatima3558** [comment]
getting the response back was taking a long time, skimmed over [this stack overflow answer](https://stackoverflow.com/a/62599037) and tried setting the headers suggested in the heading that reads "The server only allows specific user-agent strings," which worked.

**Comment by fgregg** [comment]
looking good!

Next step is to fetch the RSS from the rss url, and try to read it with feedparser

**Comment by fatima3558** [comment]
@derekeder brief notes on next steps from our convo with Forest:

check summary obj vs summary on website; sometimes rss is missing info
1. is info different?
2. look at what we return in legistar; make sure rss info is the same for info we return from legistar

**Comment by fatima3558** [comment]
@fgregg this is ready for your review!

File: civic_scraper/platforms/granicus/site.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    def create_asset(self, entry):
+        asset_name = entry['title']
+        committee_name, asset_type, str_datetime = asset_name.split(' - ')
+        meeting_datetime = datetime.strptime(str_datetime, '%b %d, %Y %I:%M %p')
+
+        meeting_url = entry['link']
+        query_dict = parse_qs(urlparse(meeting_url).query)
+        try:
+            if 'ID' in query_dict.keys():
+                meeting_id = 'granicus_{}_{}'.format(self.granicus_instance, query_dict['ID'][0])
```
Comment (fgregg) [inline civic_scraper/platforms/granicus/site.py:28]: this should be moved to the config

File: civic_scraper/platforms/granicus/site.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            else:
```
Comment (fgregg) [inline civic_scraper/platforms/granicus/site.py:29]: this should be moved to the config

File: civic_scraper/platforms/granicus/site.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                meeting_id = 'granicus_{}_{}'.format(self.granicus_instance, query_dict['MeetingID'][0])
+        except KeyError:
+            breakpoint()
+
+        e = {'url': self.url,
+             'asset_name': asset_name,
+             'committee_name': committee_name,
+             'place': None, # in Legistar, 'place' sometimes None; ok for all granicus 'place' missing?
```
Comment (fgregg) [inline civic_scraper/platforms/granicus/site.py:37]: let's set this in the config

File: civic_scraper/platforms/granicus/site.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+             'state_or_province': None,
```
Comment (fgregg) [inline civic_scraper/platforms/granicus/site.py:38]: let's set this in the config too

**Review by fgregg — CHANGES_REQUESTED** [review]

**Comment by fatima3558** [comment]
@fgregg some updates on this branch:

In revisiting this code, I realized that the `create_asset` method was not being called from the test. on further investigation, it looks like the data that feedparser was using before the break is no longer coming in. I found [this stackoverflow answer](https://stackoverflow.com/a/57637827) that talks about the error message I was getting in the response; it sounds like some recent Python changes are causing this failed response.

There are a few suggestions in the thread, but I'm not sure what makes sense; the most direct recommendation for feedparser is to monkeypatch, but that sounds like it might have some significant consequences for the whole app.

File: civic_scraper/platforms/granicus/site.py (hunk 2 above)
Comment (fatima3558) [reply civic_scraper/platforms/granicus/site.py]: this addresses the fact that on a single granicus site, entries might have different query param names for the entry ID. I've added a comment here to clarify why this conditional is here

Comment (fgregg) [reply civic_scraper/platforms/granicus/site.py]: okay, thanks!

**Comment by fgregg** [comment]
hmm.. it looks like it [feedparser.parse](https://feedparser.readthedocs.io/en/latest/introduction.html#parsing-a-feed-from-a-string) can take a string, so maybe somthing like this.

```python
import requests
import feedparser

result = request.get('THE_URL_YOU_WANT')
feed = feedparser.parse(result.text)
```

**Comment by fatima3558** [comment]
@fgregg this is ready for your review! (the re-request review button doesn't seem to be working for me today)

**Review by fgregg — APPROVED** [review]
Looks good! Once you merge, go ahead an open a PR upstream.

**Pull request merged**